install.sh

```diff
@@ -186,10 +186,10 @@ fi
 # Apply changes if the xochitl service is active
 if systemctl --quiet is-active xochitl.service 2> /dev/null; then
     touch /tmp/rm-hacks-skip-onfailure
-    if ! systemctl --quiet restart xochitl.service 2> /dev/null; then
+    if ! systemctl --quiet restart xochitl.service 2> /dev/null || systemctl --quiet is-failed xochitl.service; then
```

Comment:
Shouldn't this be && instead of || I have not tested, but when using || is it not true that the second command won't be executed if there isn't an error with the first? When previously testing `systemctl restart xochitl && echo "Success" || echo "Failed" ` I got Success when the service itself failed to restart.

Reply:
> Changing it to `&&` would make it use the following logic, which would work, but does more work than needed: `If restarting xochitl failed, an the xochitl service is-failed, then:`

A couple things:
1. I believe I have demonstrated that `systemctl restart xochitl` will return 0 even if the `xochitl` service itself fails to restart. 
2. This means that `systemctl --quiet is-failed xochitl.service` will only be executed if the call to `systemctl` itself returns a non-zero status which does not equate to `xochitl` failing to restart.

Reply:
> is it not true that the second command won't be executed if there isn't an error with the first?

Correct, the second command would only run if the first command succeeded, which is what we want. If the first command failed, we don't need to check `is-failed` anymore, as we've already encountered an issue we consider enough to uninstall on.

Reply:
>     1. I believe I have demonstrated that `systemctl restart xochitl` will return 0 even if the `xochitl` service itself fails to restart.

Yes, but if it ever returns an error code, there is no longer a need to check if `is-failed` is true. Something still failed.

>     2. This means that `systemctl --quiet is-failed xochitl.service` will only be executed if the call to `systemctl` itself returns a non-zero status which does not equate to `xochitl` failing to restart.

This is not actually correct, if `systemctl --quiet restart xochitl.service` fails, it will result in the statement being true without running `systemctl --quiet is-failed xochitl.service`. If it succeeds,  `systemctl --quiet is-failed xochitl.service` will then be checked.


You can check this all with `! systemctl restart xochitl || systemctl is-failed xochitl && echo Y || echo N`. You'll only see `Y` or `N` if `is-failed` never ran. Which means it'll always run unless `systemctl restart xochitl` has a non-zero return code.

Reply:
i'm forgetting the `!` up front. Doh!